Fix/async choices reset behaviour

davecoates opened the pull request.

davecoates committed:
fix: handle reset with accumulatePages

davecoates committed:
feature: add cleanOnOpen prop

- Default to true so results are more likely fresh
- When dropdown is opened after initial fetch it will clear previous results and
  refetch data

fanglong-alliance commented on js-packages/@prestojs/ui-antd/src/widgets/SelectAsyncChoiceWidget.tsx:
```diff
@@ -306,6 +315,27 @@ function SelectAsyncChoiceWidget<
         retrieveOptions,
     });
+    const listRef = useRef(list);
+    listRef.current = list;
```
is this line necessary?

davecoates replied:
yeah needs to be set every render. so the first time it will be initialised and then set to same value (i just did this so type on `useRef` doesn't have to include `null`)

davecoates committed:
sys: bump antd, fix types/tests

- Require antd 4.4+ now but run tests against 4.9
- Fix various typing and weird test issues with newer antd

davecoates committed:
doc: widget doc updates
